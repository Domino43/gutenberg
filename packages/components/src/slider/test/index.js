/**
 * External dependencies
 */
import { render, screen, fireEvent } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { Slider } from '..';

describe( 'Slider', () => {
	let base;

	beforeEach( () => {
		( { container: base } = render( <Slider /> ) );
	} );

	it( 'should render correctly', () => {
		expect( base.firstChild ).toMatchSnapshot();
	} );

	it( 'should render valuetext for unit-ed values', () => {
		const { container } = render( <Slider defaultValue="50px" /> );
		expect( container.firstChild ).toMatchDiffSnapshot( base.firstChild );
	} );

	it( 'should fire blur and focus events', async () => {
		const onBlur = jest.fn();
		const onFocus = jest.fn();
		render( <Slider onBlur={ onBlur } onFocus={ onFocus } /> );
		const slider = ( await screen.findAllByRole( 'slider' ) )[ 1 ];
		fireEvent.focusIn( slider );
		expect( onFocus ).toHaveBeenCalledTimes( 1 );
		fireEvent.focusOut( slider );
		expect( onBlur ).toHaveBeenCalledTimes( 1 );
	} );

	it( 'should fire onChange', async () => {
		const onChange = jest.fn();
		render( <Slider onChange={ onChange } defaultValue="50" /> );
		const slider = ( await screen.findAllByRole( 'slider' ) )[ 1 ];
		fireEvent.focusIn( slider );
		fireEvent.keyPress( slider, { key: 'ArrowUp', code: 'ArrowUp' } );
		expect( onChange ).toHaveBeenCalledWith( '51' );
	} );
} );
