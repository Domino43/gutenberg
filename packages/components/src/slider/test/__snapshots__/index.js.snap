// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render correctly 1`] = `
.emotion-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 2px;
  cursor: pointer;
  display: block;
  height: 36px;
  max-width: 100%;
  min-width: 0;
  padding: calc(4px * 1);
  width: 100%;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0::-moz-focus-outer {
  border: 0;
}

.emotion-0::-webkit-slider-runnable-track {
  background: linear-gradient(
  			to right,
  			var( --wp-admin-theme-color, #00669b) calc( var( --progress ) ),
  			rgba(0, 0, 0, 0.1) calc( var( --progress ) )
  		);
  border-radius: 2px;
  height: 2px;
}

*:disabled.emotion-0::-webkit-slider-runnable-track {
  background: rgba(0, 0, 0, 0.1);
}

.emotion-0::-moz-range-track {
  background: linear-gradient(
  			to right,
  			var( --wp-admin-theme-color, #00669b) calc( var( --progress ) ),
  			rgba(0, 0, 0, 0.1) calc( var( --progress ) )
  		);
  border-radius: 2px;
  height: 2px;
  will-change: transform;
}

*:disabled.emotion-0::-moz-range-track {
  background: rgba(0, 0, 0, 0.1);
}

.emotion-0::-webkit-slider-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: var( --wp-admin-theme-color, #00669b);
  border: 1px solid transparent;
  border-radius: 50%;
  box-shadow: none;
  cursor: pointer;
  height: 12px;
  margin-top: -5px;
  opacity: 1;
  width: 12px;
  -webkit-transition: box-shadow ease 160ms;
  transition: box-shadow ease 160ms;
}

*:disabled.emotion-0::-webkit-slider-thumb {
  background: #717171;
  border-color: #717171;
}

.emotion-0::-moz-range-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: var( --wp-admin-theme-color, #00669b);
  border: 1px solid transparent;
  border-radius: 50%;
  box-shadow: none;
  cursor: pointer;
  height: 12px;
  margin-top: -5px;
  opacity: 1;
  width: 12px;
  -webkit-transition: box-shadow ease 160ms;
  transition: box-shadow ease 160ms;
  will-change: transform;
}

*:disabled.emotion-0::-moz-range-thumb {
  background: #717171;
  border-color: #717171;
}

.emotion-0:focus::-webkit-slider-thumb {
  box-shadow: 0 0 0 2px #fff,0 0 0 3px var( --wp-admin-theme-color, #00669b);
}

.emotion-0:focus::-moz-range-thumb {
  box-shadow: 0 0 0 2px #fff,0 0 0 3px var( --wp-admin-theme-color, #00669b);
}

<input
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="components-slider emotion-0 emotion-1"
  data-wp-c16t="true"
  data-wp-component="Slider"
  max="100"
  min="0"
  role="slider"
  style="--progress: 50%;"
  type="range"
  value="50"
/>
`;

exports[`Slider should render valuetext for unit-ed values 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,10 +1,9 @@
  <input
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
-   aria-valuetext="50px"
    class="components-slider css-78ao8c-View-getFocusBoxShadow-Slider em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Slider"
    max="100"
    min="0"
`;
